api/api.go

```diff
@@ -119,6 +123,7 @@ func New(ctx context.Context, options *Options) (*web.API, error) {
 			&filters.SelectionCriteria{},
 			&filters.PlatformAwareVisibilityFilter{},
 			&filters.PatchOnlyLabelsFilter{},
+			labels.NewForbiddenLabelOperationsFilter(options.APISettings.ProctedLabels),
```

Comment:
Just a quick comment - it might be worth registering this filter before the SelectionCriteria filter as there's no need for some filters to execute if the end result would be that the filter would cut the request eventually